### ISSUE TITLE
Strata labels as color block

### ISSUE THREAD

**Issue by ddsjoberg** [ddsjoberg/ggsurvfit#34]
The strata labels can easily become quite long and it makes the layout of the figures with the risktables look wonky.
![image](https://user-images.githubusercontent.com/26774684/183417154-e9270128-a60f-4448-808e-0e5016da85f5.png)

I asked SO how we could replace the strata labels with a color block/pill. It's possible, but the code looks a bit tricky.
https://stackoverflow.com/questions/73270476

I was hoping that this option could be wrapped into a theme. But I think due to its complexity, it will be near impossible to add it in the current themes implementation (right?).

A viable option would be to add `add_risktable(strata_as_color_block=)` argument. 

@bailliem what do you think?

**Comment by ddsjoberg** [comment]
Linking the color the strata level. 


``` r
library(ggsurvfit)
p <- 
  survfit2(Surv(time, status) ~ sex, data = df_lung) |> 
  ggsurvfit(size = 1) +
  add_censor_mark() +
  add_confidence_interval()

# find the colors used in the figure
p_build <- ggplot2::ggplot_build(p)
colors <- 
  p_build$data[[1]] |> 
  dplyr::select(colour) |> 
  dplyr::distinct() |> 
  dplyr::pull()
colors
#> [1] "#F8766D" "#00BFC4"

# try to map group ID to the data strata
# all strata should be factors and therefore can just extract the 
color_label <- 
  p_build$plot$data %>%
  dplyr::pull(strata) |> 
  levels()
color_label
#> [1] "Male"   "Female"

# i hope this will work 100% of the time!
# waht modifications could a use make the the figures that could break this connection?
colors |> setNames(color_label)
#>      Male    Female 
#> "#F8766D" "#00BFC4"
```

<sup>Created on 2022-08-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by ddsjoberg** [comment]
Gahhhhh! I implemented all three methods outlined in the SO post linked above.

Each method works well UNTIL we run `align_plots()` and the color boxes disappear 😭 not sure what to do...

**Comment by ddsjoberg** [comment]
OK, I've got something that works (that also comes with concerns).

![image](https://user-images.githubusercontent.com/26774684/183634030-139b5f55-a96f-45f1-9c1f-1efb741269ff.png)

I was able to create this by formatting each of the y-axis labels into a rectangle with `scale_y_discrete(label = function(x) "\U25AC")`.

From there, i just needed to change the font color, which i did with `theme(axis.text.y.left = element_text(color = rev(color_block_mapping))` , where color_block_mapping is a vector of colors.

But when you do this, you see this warning
```r
Warning messages:
1: Vectorized input to `element_text()` is not officially supported.
Results may be unexpected or may change in future versions of ggplot2.
```

Is there another way to change the font color of the axis labels, that is, each level gets its own color?


FYI, we can also swap out the shape to be any UTF-8 shape. eg
![image](https://user-images.githubusercontent.com/26774684/183634835-8e745632-f560-4179-b507-ea32675cba7f.png)
